### PR TITLE
refactor: List border

### PR DESCRIPTION
**Pull request by paranoidjk**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1697)

### CODE REVIEW

**Comment by paranoidjk** [comment]
@ustccjw

**Comment by ustccjw** [comment]
把 dpr: 1 的情况考虑下来可以这样写：

````
.hairline(@color, @radius: 0) when (@radius = 0) or (ispixel(@radius)) or (ispercentage(@radius)) {
  border: 1PX solid @color;
  border-radius: @radius;

  html:not([data-scale]) & {
    @media (min-resolution: 2dppx) {
      position: relative;
      border: none;
      transform: translateZ(0);

      &::before {
        content: '';
        position: absolute;
        left: 0;
        top: 0;
        width: 200%;
        height: 200%;
        border: 1PX solid @color;
        border-radius: @radius * 2;
        box-sizing: border-box;
        transform-origin: 0 0;
        transform: scale(0.5);

        @media (min-resolution: 3dppx) {
          width: 300%;
          height: 300%;
          border-radius: @radius * 3;
          transform: scale(0.33);
        }
      }
    }
  }
````

**Comment by warmhug** [comment]
image
border 1PX 设置要移到`html:not([data-scale]) & {`这里边去吧

**Comment by paranoidjk** [comment]
必须要有条件判断的，不然 not scale && dpr > 1 的时候就会出现 am-list-body 本身有个 boder-top, 再加上 after 伪类画的边框，就两条线了

**Comment by ustccjw** [comment]
设置了 border: none; 呀

**Comment by paranoidjk** [comment]
@ustccjw direction  = xxx 的里面都没写

File: components/style/mixins.less [hunk 1]
```diff
@@ -155,14 +155,17 @@
...
+    }
     @media (min-resolution: 2dppx) {
       position: relative;
       transform: translateZ(0);
       &::before {
         content: '';
         position: absolute;
         left: 0;
-        top: 0;
+        top: -1PX;
```
Comment (ustccjw) [inline components/style/mixins.less]: top: 0 这样会保证在元素内部，否则会扩展到元素外部

Comment (paranoidjk) [inline components/style/mixins.less]: 在元素内部，这条线就会被内容覆盖，你测试一下？

**Comment by paranoidjk** [comment]
@ustccjw 还是 :before 上边框不显示的问题

**Comment by ustccjw** [comment]
@paranoidjk 现在应该可以了。

File: components/style/mixins.less (hunk 1 above)
Comment (ustccjw) [inline components/style/mixins.less]: 恩 改成 -1PX

**Pull request merged**